scripts/repo-health-script.sh

```diff
@@ -0,0 +1,179 @@
+pip install -q -e edx-repo-health
+
+# data destination folder setup
+
+METADATA_FILE_DIST="${WORKSPACE}/repo-health-data/docs/checks_metadata.yaml"
+
+failed_repos=()
+
+OUTPUT_FILE_POSTFIX="_repo_health.yaml"
+
+# Git clone each repo in org and run checks on it
+input="repositories.txt"
+COUNTER=0
+while IFS= read -r line
+do
+  cd $WORKSPACE
+  if [[ "${line}" =~ ^(git@github\.com:|https://github\.com/)([a-zA-Z0-9_.-]+?)/([a-zA-Z0-9_.-]+?)\.git$ ]]; then
+      ORG_NAME="${BASH_REMATCH[2]}"
+      REPO_NAME="${BASH_REMATCH[3]}"
+      FULL_NAME="${ORG_NAME}/${REPO_NAME}"
+  else
+      echo "Skipping <${line}>: Could not recognize as a GitHub URL in order to extract org and repo name."
+      continue
+  fi
+
+  if [[ "${REPO_NAME}" = "edx-repo-health" ]]; then
+      echo "Skipping <${line}>: edx-repo health"
+      continue
+  fi
+
+  if [[ -n "${ONLY_CHECK_THIS_REPOSITORY}" && "${FULL_NAME}" != "${ONLY_CHECK_THIS_REPOSITORY}" ]]; then
+      echo "Skipping <${line}>: ONLY_CHECK_THIS_REPOSITORY was set, and does not match"
+      continue
+  fi
+
+  echo "Processing repo: ${FULL_NAME}"
+
+  rm -rf target-repo
+  git clone -- "${line/https:\/\//https:\/\/$GITHUB_TOKEN@}" target-repo || {
+      failed_repos+=("$FULL_NAME")
+      continue
+  }
+  echo "Cloned repo: ${FULL_NAME}"
+  cd target-repo
+  echo "Stepping into target-repo"
+
+  # If the REPORT_DATE variable is set and not an empty string.
+  if [[ ! -z $REPORT_DATE ]]
+    then
+      # If a specific date is given for report
+      FIRST_COMMIT=$(git log --reverse --format="format:%ci" | sed -n 1p)
+      if [[ $REPORT_DATE > ${FIRST_COMMIT:0:10} ]]
+        then
+          git checkout `git rev-list -n 1 --before="${REPORT_DATE} 00:00" master`
+        else
+          echo "${REPO_NAME} doesn't have any commits prior to ${REPORT_DATE}"
+          failed_repos+=("$FULL_NAME")
+          continue
+      fi
+  fi
+
+  ORG_DATA_DIR="${WORKSPACE}/repo-health-data/individual_repo_data/${ORG_NAME}"
+  # make sure destination folder exists
+  mkdir -p "$ORG_DATA_DIR"
+
+  OUTPUT_FILE_NAME=${REPO_NAME}${OUTPUT_FILE_POSTFIX}
+  if pytest --repo-health --repo-health-path ${WORKSPACE}/edx-repo-health --repo-path ${WORKSPACE}/target-repo --repo-health-metadata "${METADATA_FILE_DIST}" --output-path "${ORG_DATA_DIR}/${OUTPUT_FILE_NAME}" -o log_cli=true --exitfirst --noconftest -v -c /dev/null
+  then
+      true
+  elif pytest --repo-health --repo-health-path ${WORKSPACE}/edx-repo-health --repo-path ${WORKSPACE}/target-repo --repo-health-metadata "${METADATA_FILE_DIST}" --output-path "${ORG_DATA_DIR}/${OUTPUT_FILE_NAME}" -o log_cli=true --exitfirst --noconftest -v -c /dev/null
+  # rerun the same command if it fails once
+  then
+      true
+  else
+      failed_repos+=("$FULL_NAME") && continue
+  fi
+  
+  # This counter will be removed after completing the testing of the script.
+  COUNTER=$((COUNTER+1))
+  echo "COUNTER : ${COUNTER}"
+  if [[ COUNTER -eq 5 ]]; then
+    break
+  fi
```

Comment:
Don't forget to remove this before merging! :)

Reply:
Removed this loop and other variables added to enable quick script execution for testing.